keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -151,3 +156,167 @@ def get_config(self):
             }
         )
         return config
+
+    def augment_bounding_boxes(self, image, bounding_boxes, transformation=None):
+        """args: image : takes a single image H,W,C,
+        bounding_boxes: take bbox coordinates [N,4] -> [y1,x1,y2,x2]
+        transformation: takes tuple for x,y transformation None if no transformation"""
+        height, width, _ = image.shape
+        input_dtype = bounding_boxes.dtype
+        image = tf.expand_dims(image, axis=0)
+        if self.bounding_box_format is None:
+            raise ValueError("Specify bounding_box_format,cannot be None")
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            images=image,
+            dtype=tf.float32,
+        )
+        x, y = transformation
+        # apply horizontal shear
+        if x is not None:
+            bounding_boxes = self._apply_horizontal_transformation_to_bounding_box(
+                bounding_boxes, x
+            )
+        # apply vertical shear
+        if y is not None:
+            bounding_boxes = self._apply_vertical_transformation_to_bounding_box(
+                bounding_boxes, y
+            )
+        # clip bounding boxes value to 0-image height and 0-image width
+        bounding_boxes = RandomShear.clip_bounding_box(height, width, bounding_boxes)
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes,
+            source="xyxy",
+            target=self.bounding_box_format,
+            images=image,
+            dtype=input_dtype,
```

Comment:
sorry, there was some confusion here.  You don't need to track input_dtype.  We decided to standardize on `self.compute_dtype` as the bounding_box output type for now.